mysite/views.py

```diff
@@ -1,3 +1,175 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect, get_object_or_404
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login, get_user_model
+from django.http import HttpResponseRedirect
+from django import forms
+from django.urls import reverse
+from django.views.generic import UpdateView
+from django.contrib import messages
+from django.contrib.auth import update_session_auth_hash
+from django.contrib.auth.forms import PasswordChangeForm
+from mysite.forms import UserRegistrationForm, UserProfileForm, ProjectForm, ProjectBidForm, AssignProjectForm
+from mysite.models import Project, ProjectBid
 
-# Create your views here.
+User = get_user_model()
+
+
+def home(request):
+    return render(request, 'mysite/home.html')
+
+
+def register(request):
+    if request.method == 'POST':
+        form = UserRegistrationForm(request.POST)
+        if form.is_valid():
+            user_obj = form.cleaned_data
```

Comment:
Please use form.save() and move all logic under post into forms save method. The view should contain minimum code for business logic. All code should be in some helper method or forms. View should only call these related method and perform decisions on the returned outputs

Reply:
Feedback Incorporated.